pkgs/applications/emulators/pcsx2/default.nix

```diff
@@ -1,90 +1,111 @@
-{ alsa-lib
-, cmake
+{ cmake
 , fetchFromGitHub
+, lib
+, stdenv
+, curl
+, ffmpeg
 , fmt
 , gettext
-, glib
-, gtk3
 , harfbuzz
-, lib
 , libaio
+, libbacktrace
 , libpcap
-, libpng
 , libpulseaudio
 , libsamplerate
-, libXdmcp
-, openssl
-, pcre
-, perl
+, libXrandr
+, libzip
 , pkg-config
-, portaudio
+, qtbase
+, qtsvg
+, qttools
+, qttranslations
+, qtwayland
+, rapidyaml
 , SDL2
 , soundtouch
-, stdenv
-, udev
 , vulkan-headers
 , vulkan-loader
-, wrapGAppsHook
-, wxGTK
-, zlib
 , wayland
+, wrapQtAppsHook
+, xz
+, zip
 }:
 
+let
+  # The pre-zipped files in releases don't have a versioned link, we need to zip them ourselves
+  cheats = fetchFromGitHub {
+    owner = "PCSX2";
+    repo = "pcsx2_patches";
+    rev = "56c088374ad398daff702a2078ce046fc5ec6787";
+    sha256 = "sha256-nZbYvJ1wt7Lm6bGHRze7i623QpW48Eon5/5saKJ/yFI=";
+  };
+in
 stdenv.mkDerivation rec {
   pname = "pcsx2";
-  version = "1.7.3331";
-  # nixpkgs-update: no auto update
+  version = "1.7.4517";
 
   src = fetchFromGitHub {
     owner = "PCSX2";
     repo = "pcsx2";
     fetchSubmodules = true;
     rev = "v${version}";
-    hash = "sha256-0RcmBMxKj/gnkNEjn2AUSSO1DzyNSf1lOZWPSUq6764=";
+    sha256 = "sha256-TwOhjtyzcvWjgtbIJu+k7RvsrhP2dUtM5uDncQFUnGM=";
   };
 
   cmakeFlags = [
     "-DDISABLE_ADVANCE_SIMD=TRUE"
-    "-DDISABLE_PCSX2_WRAPPER=TRUE"
-    "-DPACKAGE_MODE=TRUE"
-    "-DWAYLAND_API=TRUE"
-    "-DXDG_STD=TRUE"
-    "-DUSE_VULKAN=TRUE"
+    "-DUSE_SYSTEM_LIBS=ON"
+    "-DDISABLE_BUILD_DATE=TRUE"
   ];
 
-  nativeBuildInputs = [ cmake perl pkg-config vulkan-headers wrapGAppsHook ];
+  nativeBuildInputs = [ cmake pkg-config wrapQtAppsHook zip ];
 
   buildInputs = [
-    alsa-lib
+    curl
+    ffmpeg
     fmt
     gettext
-    glib
-    gtk3
     harfbuzz
     libaio
+    libbacktrace
     libpcap
-    libpng
     libpulseaudio
     libsamplerate
-    libXdmcp
-    openssl
-    pcre
-    portaudio
+    libXrandr
+    libzip
+    qtbase
+    qtsvg
+    qttools
+    qttranslations
+    qtwayland
+    rapidyaml
     SDL2
     soundtouch
-    udev
+    vulkan-headers
     vulkan-loader
     wayland
-    wxGTK
-    zlib
+    xz
   ];
 
-  preFixup = ''
-    gappsWrapperArgs+=(
-      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ vulkan-loader ]}
-    )
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -a bin/pcsx2-qt bin/resources $out/bin/
+
+    install -Dm644 $src/pcsx2/Resources/AppIcon64.png $out/share/pixmaps/PCSX2.png
+    install -Dm644 $src/.github/workflows/scripts/linux/pcsx2-qt.desktop $out/share/applications/PCSX2.desktop
+
+    zip -jq $out/bin/resources/cheats_ni.zip ${cheats}/cheats_ni/*
+    zip -jq $out/bin/resources/cheats_ws.zip ${cheats}/cheats_ws/*
   '';
 
+  qtWrapperArgs = [
+    "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [
+      ffmpeg # It's loaded with dlopen
```

Comment:
How many calls are there? Are there just a few we could patch?

Reply:
It's used here: https://github.com/PCSX2/pcsx2/blob/8de0eb3c063668493a316da95e6a5c2c56840253/pcsx2/GS/GSCapture.cpp#L219.
They plan to change it: `https://github.com/PCSX2/pcsx2/issues/8624`.

Reply:
Can we patch that instead of using LD_LIBRARY_PATH?

Reply:
As I said, they plan to change it.
Is it that bad to use `LD_LIBRARY_PATH`? It's also used on `libpulseaudio` and `vulkan-loader` after all...

Reply:
> Is it that bad to use LD_LIBRARY_PATH?

Generally it should be avoided as much as possible in as many places as possible as it taints child processes.